stdlib/observability/src/main/java/org/ballerinalang/observe/nativeimpl/OpenTracerBallerinaWrapper.java

```diff
@@ -137,13 +138,21 @@ public boolean finishSpan(int spanId) {
      * @param tagKey   the key of the tag
      * @param tagValue the value of the tag
      * @param spanId   id of the Span
+     * @param context  native context
      * @return boolean to indicate if tag was added to the span
      */
-    public boolean addTag(String tagKey, String tagValue, int spanId) {
+    public boolean addTag(String tagKey, String tagValue, int spanId, Context context) {
         if (!enabled) {
             return false;
         }
         ObserverContext observerContext = observerContextList.get(spanId);
+        if (spanId == -1) {
```

Comment:
Shall we make the `spanId` a defaultable parameter with the default value as -1? Similar to `startSpan`